add xls file suport

ic-xu opened the pull request:
# Description

We found that files in xls format were not supported for uploading to the knowledge base, so we started writing a compatible branch code, hoping to contribute to the community.


## Type of Change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Directly upload the xls file for analysis

Commit:
feat:Add parsing support for xls files

Commit:
feat:Add parsing support for xls files

crazywoola commented:
@Weaxs Please take a look at this as well.

crazywoola commented on web/yarn.lock:
Please revert this file since there are no changes in frontend part.

crazywoola reviewed (changes requested):
Hello thank you for your contribution, you can add my wechat: crazyphage
I will invite you to our contributor's group.

Commit:
doc:Modify comments to English

Weaxs commented on api/core/rag/extractor/excel_extractor.py:
```diff
@@ -27,10 +28,37 @@ def __init__(
+
+    def _extract4xls(self) -> list[Document]:
+        wb = xlrd.open_workbook(filename=self._file_path)
+        row_header = None
```
every sheet needs to initialize  row_header， maybe move this to line 43 after `for sheet in wb.sheets():`

ic-xu replied:
I'm sorry for not thinking of this part, he did have such logic, I will fix it

Weaxs commented on api/core/rag/extractor/excel_extractor.py:
```diff
@@ -27,10 +28,37 @@ def __init__(
-
-        # 使用 Pandas 读取 Excel 文件的每个工作表
+        # Read each worksheet of an Excel file using Pandas
         xls = pd.ExcelFile(self._file_path)
```
`pd.ExcelFile(self._file_path)` can read xlsx and xls, It's not necessary to add functions `_extract4xls` and `_extract4xlsx` ?

ic-xu replied:
Yes, we need it. Pandas cannot read media information. We may need to read media information later.

Weaxs reviewed (commented):
It's not necessary to change `excel_extractor.py`, just add `.xls'` type in `extract_processor.py`  and `file_service.py`

bowenliang123 commented on web/yarn.lock:
leave it untouched, since no npm dependencies updated.

Commit:
fix:revert yarn.lock

Commit:
Revert "chore(deps): bump katex from 0.16.8 to 0.16.10 in /web (#3307)"

This reverts commit 81e1b3fc61ca4210e012faf91474816e0296a9ae.

Commit:
1、fix:revert yarn.lock;2、fix:every sheet needs to initialize row_header， maybe move this to line 43 after for sheet in wb.sheets():,
in   def _extract4xls(self) -> list[Document]:
        wb = xlrd.open_workbook(filename=self._file_path)
        row_header = None

Commit:
feat: merge main

Commit:
fix: merge conflicts
